### PR TITLE
feat: annotations edit modal

### LINKED ISSUE #896: Annotations: On Annotation Click
### Background:
We need the ability to interact with the annotation. Clicking on it should launch a modal:

See #899 #900

### Acceptance Criteria
- ~Clicking on a spot in the graph that is already occupied by an annotation will launch a dialog similar to the one in the screenshot above~ 
- Annotations will have a marker / target for easy clicking

### LINKED ISSUE #951: Annotations: Wire up Edit / Delete overlay
### Background
In order to break things up and keep things moving, we should wire up the layout and design of the edit / delete annotation overlay. Once we get the affordance for editing an annotation, we can hook this overlay up to that action.

### PR DESCRIPTION
**Pull request by ShmuelLotman**
Closes #951 

This PR introduces an edit modal for annotations when clicking on an annotation line. Currently, the functions to delete, edit are no-ops, but this will be addressed in #899 and #900
- [x] [E2E Tests pass in K8S-IDPE](https://docs.influxdata.io/development/guides/local-development/#three-ways-to-run-the-ui-e2e-tests-against-the-kind-cluster) (applicable only if you edited Cypress tests)

### CODE REVIEW

File: src/annotations/components/EditAnnotationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                size={ComponentSize.Medium}
+              />
+            </Form.Element>
+            <Form.Element
+              label="Message"
+              className="edit-annotation-form-label"
+            >
+              <TextArea
+                name="message"
+                placeholder="Try writing markdown here..."
```
Comment (hoorayimhelping) [inline src/annotations/components/EditAnnotationForm.tsx:110]: don't think we need this, do we?

File: src/annotations/components/EditAnnotationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                  >
+                    {annotationStreams[0].stream}
+                  </Dropdown.Button>
+                )}
+                menu={onCollapse => (
+                  <Dropdown.Menu
+                    onCollapse={onCollapse}
+                    testID="stream-selector--dropdown-menu"
+                  >
+                    {annotationStreams.map(a => a.stream)}
```
Comment (hoorayimhelping) [inline src/annotations/components/EditAnnotationForm.tsx:144]:
```suggestion
                    {annotationStreams.map(stream => stream.stream)}
```

File: src/visualization/types/Graph/view.tsx [hunk 3]
```diff
@@ -174,6 +174,12 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
     return <EmptyGraphMessage message={INVALID_DATA_COPY} />
   }
 
+  const editAnnotation = (userSelectedAnnotation: Annotation) => {
+    event('xyplot.annotations.edit_annotation.edit')
+    /* eslint-disable no-console */
+    console.log('edit!', userSelectedAnnotation)
```
Comment (hoorayimhelping) [inline src/visualization/types/Graph/view.tsx:180]: ✂️

File: src/visualization/types/Graph/view.tsx [hunk 4]
```diff
@@ -174,6 +174,12 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
     return <EmptyGraphMessage message={INVALID_DATA_COPY} />
   }
 
+  const editAnnotation = (userSelectedAnnotation: Annotation) => {
+    event('xyplot.annotations.edit_annotation.edit')
```
Comment (hoorayimhelping) [inline src/visualization/types/Graph/view.tsx:178]: nice!

File: src/visualization/types/Graph/view.tsx [hunk 5]
```diff
@@ -192,20 +198,43 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
...
+        dispatch(
+          showOverlay(
+            'edit-annotation',
+            {editAnnotation, clickedAnnotation},
+            () => {
+              event('xyplot.annotations.edit_annotation.cancel')
+              dismissOverlay()
+            }
+          )
+        )
```
Comment (hoorayimhelping) [inline src/visualization/types/Graph/view.tsx:221]: i think this needs to be removed right? since we're changing how we launch the annotation popup

File: src/visualization/types/Graph/view.tsx [hunk 6]
```diff
@@ -276,6 +298,12 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
       }
     })
 
+    if (inAnnotationWriteMode) {
+      config.interactionHandlers = {
+        singleClick: makeSingleClickHandler(),
+      }
+    }
```
Comment (hoorayimhelping) [inline src/visualization/types/Graph/view.tsx:305]: it doesn't seem to affect things, but why this change in position?

**Review by hoorayimhelping — CHANGES_REQUESTED** [review]
Good work, have a couple of little nitpicks and tips on the best way to use clockface.

I updated the description to point to this ticket: https://github.com/influxdata/ui/issues/951, which I just wrote up

File: src/visualization/types/Graph/view.tsx (hunk 5 above)
Comment (ShmuelLotman) [reply src/visualization/types/Graph/view.tsx:221]: Once we actually come up with that work and it gets merged in, then i would remove this. But as of  now, to get this working and unblock work on edit/delete, id leave this here so we can focus on those

File: src/annotations/components/EditAnnotationForm.tsx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        <div className="edit-annotation-buttons">
+          <Button
+            text="Cancel"
+            onClick={onClose}
+            color={ComponentColor.Default}
+            className="edit-annotation-cancel"
+          />
+          <Button
+            text="Save Changes"
+            onClick={(e: FormEvent) =>
```
[the remark this answers is not in the document]
Comment (ShmuelLotman) [reply src/annotations/components/EditAnnotationForm.tsx:168]: Yeah this was a holdover from some other funky stuff - i'll give you a bit of context, was a fun ride - you can't use the callback form of useState with the [e.target.name]: [e.target.value] because React loses context to the synthetic event.... so was playing around with that for a bit - just thought its a cool tidbit to share

File: src/visualization/types/Graph/view.tsx (hunk 6 above)
Comment (ShmuelLotman) [reply src/visualization/types/Graph/view.tsx:305]: Just helped me keep track of this as i was reading

File: src/visualization/types/Graph/view.tsx (hunk 3 above)
Comment (ShmuelLotman) [reply src/visualization/types/Graph/view.tsx:180]: I want to leave this here to have the signature stay the same and to prevent me from having to add more typescript rules just to leave the unused var in place lol - you can thank Typescript's obnoxious yelling

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 33 | :red_circle:
accessibility | 96 | :green_circle:
best-practices | 93 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :question:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=943)

**Review by hoorayimhelping — APPROVED** [review]
